src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenForEachTests.cs

```diff
@@ -1352,6 +1352,326 @@ class NonDisposableEnumerator : AbstractEnumerator
 -3");
         }
 
+        [Fact]
+        public void TestForEachPatternDisposableRefStruct()
+        {
+            var source = @"
+class C
+{
+    static void Main()
+    {
+        foreach (var x in new Enumerable1())
+        {
+            System.Console.WriteLine(x);
+        }
+    }
+}
+
+class Enumerable1
+{
+    public DisposableEnumerator GetEnumerator() { return new DisposableEnumerator(); }
+}
+
+ref struct DisposableEnumerator
+{
+    int x;
+    public int Current { get { return x; } }
+    public bool MoveNext() { return ++x < 4; }
+    public void Dispose() { System.Console.WriteLine(""Done with DisposableEnumerator""); }
+}";
+            var compilation = CompileAndVerify(source, expectedOutput: @"
+1
+2
+3
+Done with DisposableEnumerator").VerifyIL("C.Main", @"
```

Comment:
Please add a comment indicating this helps verify we are not accidentally boxing the enumerator type. It's not obvious at a glance that this is the purpose.

Reply:
May want to link to the CoreCLR bug here as well to provide more context.

---
In reply to: [250429919](https://github.com/dotnet/roslyn/pull/32640#discussion_r250429919) [](ancestors = 250429919)